### PR TITLE
azurerm_cosmosdb_account: add new capabilities EnableAggregationPipeline, MongoDBv3.4, mongoEnableDocLevelTTL

### LINKED ISSUE #1848: Add Support for EnableAggregationPipeline Capability in Cosmos DB (Mongo)
### Description

Cosmos DB has added a preview feature to enable aggregation pipelines to its implementation of Mongo. In Cosmos parlance this feature is called `EnableAggregationPipeline`

### New or Affected Resource(s)

* azurerm_cosmosdb_account

### Potential Terraform Configuration

```hcl
resource "azurerm_cosmosdb_account" "db" {
  name                = "my-cosmos-db"
  location            = "${azurerm_resource_group.rg.location}"
  resource_group_name = "${azurerm_resource_group.rg.name}"
  offer_type          = "Standard"
  kind                    = "MongoDB"

  enable_automatic_failover = false
  capabilities        = [{
    name = "EnableAggregationPipeline"
  }]
}
```

### References

More information about this capability can be found here:

* https://azure.microsoft.com/en-us/updates/azure-documentdb-now-supports-aggregate-functions/

### PR DESCRIPTION
**Pull request by katbyte**
fixes #1848 #1927

uses workaround ad described in https://github.com/Azure/azure-sdk-for-go/issues/2864

### CODE REVIEW

**Comment by katbyte** [comment]
```
[12:57:22] kt@snowbook:~/hashi/..3../terraform-providers/terraform-provider-azurerm▸f-cosmos-mongo34$ testazure TestAccAzureRMCosmosDBAccount_capability
==> Fixing source code with gofmt...
# This logic should match the search logic in scripts/gofmtcheck.sh
gofmt -s -w `find . -name '*.go' | grep -v vendor`
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./azurerm -v -test.run=TestAccAzureRMCosmosDBAccount_capability -timeout 180m -ldflags="-X=github.com/terraform-providers/terraform-provider-azurerm/version.ProviderVersion=acc"
=== RUN   TestAccAzureRMCosmosDBAccount_capabilityGremlin
=== PAUSE TestAccAzureRMCosmosDBAccount_capabilityGremlin
=== RUN   TestAccAzureRMCosmosDBAccount_capabilityTable
=== PAUSE TestAccAzureRMCosmosDBAccount_capabilityTable
=== RUN   TestAccAzureRMCosmosDBAccount_capabilityCassandra
=== PAUSE TestAccAzureRMCosmosDBAccount_capabilityCassandra
=== RUN   TestAccAzureRMCosmosDBAccount_capabilityAggregationPipeline
=== PAUSE TestAccAzureRMCosmosDBAccount_capabilityAggregationPipeline
=== RUN   TestAccAzureRMCosmosDBAccount_capabilityMongo35
=== PAUSE TestAccAzureRMCosmosDBAccount_capabilityMongo35
=== RUN   TestAccAzureRMCosmosDBAccount_capabilityDocLevelTTL
=== PAUSE TestAccAzureRMCosmosDBAccount_capabilityDocLevelTTL
=== CONT  TestAccAzureRMCosmosDBAccount_capabilityGremlin
=== CONT  TestAccAzureRMCosmosDBAccount_capabilityMongo35
=== CONT  TestAccAzureRMCosmosDBAccount_capabilityCassandra
=== CONT  TestAccAzureRMCosmosDBAccount_capabilityDocLevelTTL
=== CONT  TestAccAzureRMCosmosDBAccount_capabilityAggregationPipeline
=== CONT  TestAccAzureRMCosmosDBAccount_capabilityTable
--- PASS: TestAccAzureRMCosmosDBAccount_capabilityMongo35 (1321.83s)
--- PASS: TestAccAzureRMCosmosDBAccount_capabilityGremlin (1321.91s)
--- PASS: TestAccAzureRMCosmosDBAccount_capabilityCassandra (1323.35s)
--- PASS: TestAccAzureRMCosmosDBAccount_capabilityDocLevelTTL (1338.59s)
--- PASS: TestAccAzureRMCosmosDBAccount_capabilityAggregationPipeline (1339.40s)
--- PASS: TestAccAzureRMCosmosDBAccount_capabilityTable (1421.25s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm	1422.289s
[13:42:28] kt@snowbook:~/hashi/..3../terraform-providers/terraform-provider-azurerm▸f-cosmos-mongo34$ gitacp "patch capabilities"
```

File: azurerm/resource_arm_cosmos_db_account.go [hunk 1]
```diff
@@ -370,6 +373,22 @@ func resourceArmCosmosDBAccountCreate(d *schema.ResourceData, meta interface{})
...
+	future, err := client.Patch(ctx, resourceGroup, name, documentdb.DatabaseAccountPatchParameters{
+		DatabaseAccountPatchProperties: &documentdb.DatabaseAccountPatchProperties{
+			Capabilities: account.Capabilities,
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("Error Patching CosmosDB Account %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	future.WaitForCompletionRef(context.Background(), client.Client)
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmos_db_account.go]: We should assign err here

File: azurerm/resource_arm_cosmos_db_account.go [hunk 2]
```diff
@@ -370,6 +373,22 @@ func resourceArmCosmosDBAccountCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Error creating CosmosDB Account %q (Resource Group %q): %+v", name, resourceGroup, err)
 	}
 
+	//for some reason capabilities doesn't always work on create, so lets patch it
+	//tracked: https://github.com/Azure/azure-sdk-for-go/issues/2864
+	future, err := client.Patch(ctx, resourceGroup, name, documentdb.DatabaseAccountPatchParameters{
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmos_db_account.go]: Do we want this in the update method too?

File: azurerm/resource_arm_cosmos_db_account_test.go [hunk 3]
```diff
@@ -678,6 +778,46 @@ func testAccAzureRMCosmosDBAccount_table(rInt int, location string) string {
     `)
 }
 
+func testAccAzureRMCosmosDBAccount_capabilityCassandra(rInt int, location string) string {
+	return testAccAzureRMCosmosDBAccount_basic(rInt, location, string(documentdb.BoundedStaleness), "", `
+        kind = "GlobalDocumentDB"
+
+        capabilities = {
+          name = "EnableCassandra"
+        }
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmos_db_account_test.go]: It’d be good to add a test handling an update here

File: azurerm/resource_arm_cosmos_db_account.go [hunk 4]
```diff
@@ -200,7 +200,10 @@ func resourceArmCosmosDBAccount() *schema.Resource {
 							ValidateFunc: validation.StringInSlice([]string{
 								"EnableTable",
 								"EnableGremlin",
-								`EnableCassandra`,
+								"EnableCassandra",
+								"EnableAggregationPipeline",
+								"MongoDBv3.4",
+								"mongoEnableDocLevelTTL",
```
Comment (tombuildsstuff) [inline azurerm/resource_arm_cosmos_db_account.go]: docs are missing for these - but I'll push a commit for this

**Pull request merged**